syntax = "proto3";
package protoBuf;
import "google/protobuf/any.proto";

// To compile the protobuf to Go run the following inside the protoBuf folder:
// protoc -I=. --go_out=plugins=grpc:.  computation.proto

// To compile the protobuf to Python run the following inside the protoBuf folder:
// python -m grpc_tools.protoc -I=. --python_out=. --grpc_python_out=. computation.proto

// Message relayed by an algo to coordinator and site-connector that contains the query to be performed and algo id
message ComputeRequest {
    int32 algo_id = 1;
    google.protobuf.Any req = 2;
}

message ComputeResponse {
    google.protobuf.Any response = 1;
}

message ComputeResponses {
    repeated ComputeResponse responses = 1;
}

message IntResponse {
    int32 val = 1;
}

// Simple example query
message Query {
    string operator = 1;
    string field = 2;
    string string_value = 3;
    int32  numeric_value = 4;
}

message Patient {
    string first_name = 1;
    string last_name = 2;
    string email = 3;
    int32 age = 4;

    enum Gender {
        MALE = 0;
        FEMALE = 1;
        OTHER = 2;
    }

    Gender gender = 5;
    float weight = 6;  // Weight in kilograms
    int32 height = 7;  // Height in centimeters
}